stablehlo/tests/infer_stablehlo.mlir

```diff
@@ -275,6 +275,34 @@ func.func @batch_norm_inference(%input: tensor<4x256xf32>, %scale: tensor<256xf3
   func.return %1 : tensor<4x256xindex>
 }
 
+// -----
+
+// CHECK-LABEL: if 
+func.func @if(%pred : tensor<i1>, %branch_operand : tensor<2xf32>, %wrong_type : tensor<2xf32>) {
+  %0 = "stablehlo.if"(%pred) ({
+      "stablehlo.return"(%wrong_type) : (tensor<2xf32>) -> ()
```

Comment:
What kind of terminators are we expecting in conditional ops? I think we should restrict them to just `stablehlo.return` to make sure we don't depend on other dialects (so that we can provide robust compatibility guarantees).

Reply:
Looks like all_reduce/reduce_scatter/map all have this problem. Could you explain the solution in your mind? I'm thinking of a check like: `region->front().getTerminator().isa<ReturnOp>()` but what the terminator in negative test looks like?
these are not valid:
"return"(%branch_operand) : (tensor<2xf32>) -> ()
"llvm.return"(%branch_operand) : (tensor<2xf32>) -> ()

Reply:
Huh you're right. Looks like this is already validated for us :)